src/main/java/com/google/flourbot/api/DriveCloudSheet.java

```diff
@@ -10,51 +10,104 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Collections;
 
 public class DriveCloudSheet implements CloudSheet {
-  Sheets sheetsService;
-  String spreadsheetId;
+  private static final String VALUE_INPUT_OPTION = "USER_ENTERED";
+  private static final String INSERT_DATA_OPTION = "INSERT_ROWS";
+  private Sheets sheetsService;
+  private String spreadsheetId;
 
   public DriveCloudSheet(String documentId, Sheets documentService) {
     this.spreadsheetId = documentId;
     this.sheetsService = documentService;
   }
 
-  // TODO: replace hardcoded range with calculated range (so far values must be length 3)
-  public void appendRow(List<String> values) throws IOException, GeneralSecurityException {
+  private String getRange(int numEntryColumns, int numSkipColumns) {
+    // Returns range for sheets request by calculating the alphabetic name of
+    // the start and end column based on the number of columns filled and pre-skipped
+
+    String startColumn = toAlphabetic(numSkipColumns);
+    String endColumn = toAlphabetic(numSkipColumns + numEntryColumns - 1);
+    return String.format("Sheet1!%s:%s", startColumn, endColumn);
+  }
+
+  private String toAlphabetic(int i) {
+    // Recursive strategy to build a string that represents the alphabetic name
+    // of a column based on column number i
+    // e.g. 0 -> A, 1 -> B... 25 -> Z, 26 -> AA...
     
-    // Generate request to append to sheet
-    String range = "Sheet1!A1:C1";
-    String valueInputOption = "USER_ENTERED";
-    String insertDataOption = "INSERT_ROWS";
+    // Negative numbers cause errors
+    if (i < 0) {
+      throw new IllegalArgumentException(String.format("Column number %d is invalid", i));
+    }
+
+    // Every calculation produces 1 letter
+    int quotient = i / 26;
+    char letter = (char)(i % 26 + 'A');
```

Comment:
I still think this should be 27, can you convince me?

Reply:
A->Z is 26 letters, so if we do i % 26, we have results [0,25] which is 26 different numbers. This gives us A + 0 = A... A + 25 = Z

Reply:
Sorry, yeah I got confused because I assumed that the columns will be indexed from 1... it is still not clear to me that they won't, so it would be good to make it clear in javadoc of all the methods that use it.